### PR TITLE
DROOLS-1492 typeRef allowed values check for DT input and output clauses, and InputData

### PR DESCRIPTION
**Pull request by tarilabs**
@etirelli kindly let me know if the check on the `InputData` shall be part of this same DROOLS-1492 task or we can manage it separate.

### CODE REVIEW

**Comment by etirelli** [comment]
@tarilabs I think it can be all under the same ticket, no problem.

**Comment by tarilabs** [comment]
Commit 22ce276 adds the check for InputData typeRef allowedValues as well.

File: kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNFEELHelper.java [hunk 1]
```diff
@@ -44,6 +48,32 @@ private FEEL createFEELInstance() {
...
+    
+            for ( UnaryTest t : unaryTests ) {
+                try {
+                    Boolean applyT = t.apply(ctx, value);
+                    if ( applyT ) {
+                        return true;
+                    }
+                } catch ( Throwable e ) {
+                    // if an unaryTest fail for any treason, is simply considered FALSE.
+                    logger.error("An error happended while evaluating a unary test", e);
```
Comment (etirelli) [inline kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNFEELHelper.java]: Sorry for changing my mind, but I think I agree with you. Lets downgrade this to a logger.warn() and change the message to: "A non-critical error happened while evaluating a unary test. Execution will continue.".

**Comment by etirelli** [comment]
+1, lets merge it.

File: kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNFEELHelper.java (hunk 1 above)
Comment (tarilabs) [inline kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNFEELHelper.java]: Done with commit 7c0c384

**Pull request merged**